### PR TITLE
nheko: Add recipe and patchset for 0.11.3

### PR DESCRIPTION
**Pull request by m199**
Tested on hrev57257

### CODE REVIEW

File: net-im/nheko/nheko-0.11.3.recipe [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+	devel:libssl$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	devel:nlohmann_json
+	"
+BUILD_PREREQUIRES="
+	cmd:g++$secondaryArchSuffix
+	cmd:cmake
+	cmd:linguist$secondaryArchSuffix >= 5
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
```
Comment (korli) [inline net-im/nheko/nheko-0.11.3.recipe:102]:
```suggestion
	cmd:cmake
	cmd:g++$secondaryArchSuffix
	cmd:linguist$secondaryArchSuffix >= 5
	cmd:make
	cmd:pkg_config$secondaryArchSuffix
```

File: net-im/nheko/patches/nheko-0.11.3.patchset [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+diff --git a/src/notifications/Manager.h b/src/notifications/Manager.h
+index 70a03345..e908575e 100644
+--- a/src/notifications/Manager.h
++++ b/src/notifications/Manager.h
+@@ -55,7 +55,7 @@ public slots:
+ #if defined(NHEKO_DBUS_SYS)
+ public:
+     void closeNotifications(QString roomId);
+-#if defined(Q_OS_LINUX)
++#if defined(Q_OS_HAIKU)
```
Comment (korli) [inline net-im/nheko/patches/nheko-0.11.3.patchset:45]:
```suggestion
+#if defined(Q_OS_HAIKU) || defined(Q_OS_LINUX)
```

File: net-im/nheko/nheko-0.11.3.recipe [hunk 3]
```diff
@@ -0,0 +1,138 @@
+SUMMARY="An IM client for Matrix"
+DESCRIPTION="The motivation behind the nheko is to provide a native desktop \
+app for Matrix that feels more like a mainstream chat app (Element, Telegram \
+etc.) and less like an IRC client. It was developed using the Qt5 framework."
+HOMEPAGE="https://nheko.im/nheko-reborn/nheko"
+COPYRIGHT="2017-2023 Nheko Authors"
+LICENSE="GNU GPL v3"
+REVISION="1"
+SOURCE_URI="https://github.com/Nheko-Reborn/nheko/archive/v${portVersion}.tar.gz"
```
Comment (korli) [inline net-im/nheko/nheko-0.11.3.recipe:9]:
```suggestion
SOURCE_URI="https://github.com/Nheko-Reborn/nheko/archive/v$portVersion.tar.gz"
```

File: net-im/nheko/nheko-0.11.3.recipe [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+CHECKSUM_SHA256="f285156884a3a0c6870f3fba89c13d1fd70c8727bd179d8310b13819f8a63a37"
+SOURCE_FILENAME="nheko-${portVersion}.tar.gz"
```
Comment (korli) [inline net-im/nheko/nheko-0.11.3.recipe:11]:
```suggestion
SOURCE_FILENAME="nheko-$portVersion.tar.gz"
```

File: net-im/nheko/nheko-0.11.3.recipe [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+SOURCE_DIR="nheko-${portVersion}"
```
Comment (korli) [inline net-im/nheko/nheko-0.11.3.recipe:12]: not needed

File: net-im/nheko/nheko-0.11.3.recipe [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+PATCHES="nheko-${portVersion}.patchset"
```
Comment (korli) [inline net-im/nheko/nheko-0.11.3.recipe:13]:
```suggestion
PATCHES="nheko-$portVersion.patchset"
```

File: net-im/nheko/nheko-0.11.3.recipe [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+ARCHITECTURES="all !x86_gcc2"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	nheko$secondaryArchSuffix = $portVersion
+	app:Nheko$secondaryArchSuffix = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	gstreamer${secondaryArchSuffix}
```
Comment (korli) [inline net-im/nheko/nheko-0.11.3.recipe:25]:
```suggestion
	gstreamer$secondaryArchSuffix
```

File: net-im/nheko/nheko-0.11.3.recipe [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+	lib:libQt5Multimedia$secondaryArchSuffix
+	lib:libQt5Qml$secondaryArchSuffix
+	lib:libQt5Quick$secondaryArchSuffix
+	lib:libQt5QuickControls2$secondaryArchSuffix
+	lib:libQt5QuickWidgets$secondaryArchSuffix
+	lib:libQt5Svg$secondaryArchSuffix
+	lib:libQt5Widgets$secondaryArchSuffix
+	lib:libspdlog$secondaryArchSuffix
+	lib:libssl$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
```
Comment (korli) [inline net-im/nheko/nheko-0.11.3.recipe:56]:
```suggestion
	lib:libcmark$secondaryArchSuffix
	lib:libcoeurl$secondaryArchSuffix
	lib:libcrypto$secondaryArchSuffix
	lib:libcurl$secondaryArchSuffix
	lib:libfmt$secondaryArchSuffix
	lib:libglib_2.0$secondaryArchSuffix
	lib:libgstsdp_1.0$secondaryArchSuffix
	lib:libgstwebrtc_1.0$secondaryArchSuffix
	lib:liblmdb$secondaryArchSuffix
	lib:libmatrix_client$secondaryArchSuffix
	lib:libolm$secondaryArchSuffix
	lib:liborc_0.4$secondaryArchSuffix
	lib:libQt5Concurrent$secondaryArchSuffix
	lib:libQt5Core$secondaryArchSuffix
	lib:libQt5DBus$secondaryArchSuffix
	lib:libQT5Gui$secondaryArchSuffix
	lib:libqt5keychain$secondaryArchSuffix
	lib:libQt5Multimedia$secondaryArchSuffix
	lib:libQt5Qml$secondaryArchSuffix
	lib:libQt5Quick$secondaryArchSuffix
	lib:libQt5QuickControls2$secondaryArchSuffix
	lib:libQt5QuickWidgets$secondaryArchSuffix
	lib:libQt5Svg$secondaryArchSuffix
	lib:libQt5Widgets$secondaryArchSuffix
	lib:libre2$secondaryArchSuffix
	lib:libspdlog$secondaryArchSuffix
	lib:libssl$secondaryArchSuffix
	lib:libz$secondaryArchSuffix
```

File: net-im/nheko/nheko-0.11.3.recipe [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+	devel:libQt5Qml$secondaryArchSuffix
+	devel:libQt5Quick$secondaryArchSuffix
+	devel:libQt5QuickControls2$secondaryArchSuffix
+	devel:libQt5QuickWidgets$secondaryArchSuffix
+	devel:libQt5Svg$secondaryArchSuffix
+	devel:libQt5Widgets$secondaryArchSuffix
+	devel:libspdlog$secondaryArchSuffix
+	devel:libssl$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	devel:nlohmann_json
```
Comment (korli) [inline net-im/nheko/nheko-0.11.3.recipe:95]:
```suggestion
	devel:libcmark$secondaryArchSuffix
	devel:libcoeurl$secondaryArchSuffix
	devel:libcurl$secondaryArchSuffix
	devel:libevent$secondaryArchSuffix
	devel:libfmt$secondaryArchSuffix
	devel:libglib_2.0$secondaryArchSuffix
	devel:libgstsdp_1.0$secondaryArchSuffix
	devel:libgstwebrtc_1.0$secondaryArchSuffix
	devel:liblmdb$secondaryArchSuffix
	devel:libmatrix_client$secondaryArchSuffix  >= 0.5.1
	devel:libolm$secondaryArchSuffix
	devel:liborc_0.4$secondaryArchSuffix
	devel:libQt5Concurrent$secondaryArchSuffix
	devel:libQt5Core$secondaryArchSuffix
	devel:libQt5Dbus$secondaryArchSuffix
	devel:libQt5Gui$secondaryArchSuffix
	devel:libqt5keychain$secondaryArchSuffix
	devel:libQt5Multimedia$secondaryArchSuffix
	devel:libQt5Network$secondaryArchSuffix
	devel:libQt5Qml$secondaryArchSuffix
	devel:libQt5Quick$secondaryArchSuffix
	devel:libQt5QuickControls2$secondaryArchSuffix
	devel:libQt5QuickWidgets$secondaryArchSuffix
	devel:libQt5Svg$secondaryArchSuffix
	devel:libQt5Widgets$secondaryArchSuffix
	devel:libre2$secondaryArchSuffix
	devel:libspdlog$secondaryArchSuffix
	devel:libssl$secondaryArchSuffix
	devel:libz$secondaryArchSuffix
	devel:nlohmann_json
```

Comment (korli) [reply net-im/nheko/nheko-0.11.3.recipe:95]: is devel:libevent actually needed?

Comment (m199) [reply net-im/nheko/nheko-0.11.3.recipe:95]: Yes, without it has a build error "No package 'libevent_core' found"

File: net-im/nheko/nheko-0.11.3.recipe (hunk 5 above)
Comment (m199) [reply net-im/nheko/nheko-0.11.3.recipe:12]:
```suggestion
```

File: net-im/nheko/nheko-0.11.3.recipe (hunk 9 above)
Comment (korli) [reply net-im/nheko/nheko-0.11.3.recipe]: OK Thanks!

**Pull request merged**